pyqtorch/adjoint.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from typing import Any
+
+import torch
+
+import pyqtorch as pyq
+from pyqtorch.parametric import Parametric
+from pyqtorch.utils import overlap, param_dict
+
+
+class AdjointExpectation(torch.autograd.Function):
```

Comment:
i think i would like to have it in native pyq. we can obviously reuse the logic here easily for qadence once we figure out the details on how to deal with dagger/jacobian native methods of backends in qadence. @madagra ?

Reply:
Moving to `qadence` makes sense in the long run, I think, but I would do it later on. AFAIK, today the only backend that can implement adjoint derivatives is PyQTorch. Moving the whole logic to `qadence` now would just imply quite a lot of refactoring work without any practical benefit in the end.

Reply:
And @dominikandreasseitz please separate the refactoring of PyQTorch from the adjoint in two separate PRs.

Reply:
I am still of the opinion that `dagger/jacobian` are to be used outside the purposes of the adjoint method. Hence the point of moving it to Qadence. Another argument is that it can be used with another diff engine.

Reply:
i can only reiterate my point that the backend for which we would use the adjoint, would need to implement _native_ dagger and jacobian methods which give you the actual tensor/matrix which you then use to "unapply" the state and compute the grad in case of a parametric op. this is not the case for any other backends yet

Reply:
All right, there might be a point I fail to see here. I should be having a deeper look at the adjoint method then.

Reply:
lets move this discussion somwhere else since this MR will be split anyway.